arrows/ocv/inpaint.cxx

```diff
@@ -0,0 +1,169 @@
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "inpaint.h"
+
+#include <arrows/ocv/image_container.h>
+
+#include <vital/util/enum_converter.h>
+#include <vital/vital_config.h>
+
+#include <opencv2/core/core.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/photo.hpp>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace ocv {
+
+enum inpainting_method
+{
+  METHOD_mask,
+  METHOD_telea,
+  METHOD_navier_stokes,
+};
+
+ENUM_CONVERTER( method_converter, inpainting_method, { "mask", METHOD_mask },
+                { "telea", METHOD_telea },
+                { "navier_stokes", METHOD_navier_stokes } )
+
+// ----------------------------------------------------------------------------
+/// Private implementation class
+class inpaint::priv
+{
+public:
+
+  priv( inpaint* parent )
+    : p{ parent }
+  {
+  }
+
+  ~priv()
+  {
+  }
+
+  inpaint* p;
+  // Internal parameters/settings
+  inpainting_method method{ METHOD_telea };
+  float radius{ 3.0 };
+};
+
+// ----------------------------------------------------------------------------
+inpaint
+::inpaint()
+  : d{ new priv{ this } }
+{
+  attach_logger( "arrows.ocv.inpaint" );
+}
+
+// ----------------------------------------------------------------------------
+inpaint::~inpaint()
+{
+}
+
+// ----------------------------------------------------------------------------
+vital::config_block_sptr
+inpaint
+::get_configuration() const
+{
+  // get base config from base class
+  vital::config_block_sptr config = algorithm::get_configuration();
+
+  config->set_value( "inpaint_method", method_converter().to_string(
+                       d->method ),
+                     "Inpainting method, possible values: " +
+                     method_converter().element_name_string() );
+  config->set_value( "radius", d->radius,
+                     "Radius parameter for the inpainting method" );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+inpaint
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Starting with our generated vital::config_block to ensure that assumed
+  // values are present. An alternative is to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for filtering
+  d->method = config->get_enum_value< method_converter >( "inpaint_method",
+                                                          d->method );
+  d->radius = config->get_value< float >( "radius", d->radius );
+}
+
+// ----------------------------------------------------------------------------
+bool
+inpaint
+::check_configuration( vital::config_block_sptr config ) const
+{
+  auto const radius = config->get_value< float >( "radius" );
+  if( radius <= 0 )
+  {
+    LOG_ERROR( logger(),
+               "Radius should be positive but instead was " << radius );
```

Comment:
```suggestion
               "Radius should be non-negative but instead was " << radius );
```
...oh, and it occurs to me, is it possible to feed in a non-finite value? I believe that would be accepted by this check, though I'm not sure the config parser accepts non-finite values.

Reply:
0 is not an acceptable value as this is written, so I believe that the original message is correct. I'm not sure if it's possible to have an infinite value in a config, but I feel like you'd have to try pretty hard to mess up that badly. My intuition is anything like `inf` would get parsed as a string or fail.

Reply:
Ack, somehow I was thinking this was written in a way that would accept zero :flushed:.

Nothing to see, here. Move along.